service/collector/chart_config_resource.go

```diff
@@ -117,11 +118,16 @@ func (a *ChartConfigResource) Describe(ch chan<- *prometheus.Desc) error {
 }
 
 func convertToTime(datetime string) (time.Time, error) {
-	layout := "2006-01-02T15:04:05.000Z"
-	t, err := time.Parse(layout, datetime)
+	layout := "2006-01-02T15:04:05"
 
+	split := strings.Split(datetime, ".")
+	if len(split) == 0 {
+		return time.Time{}, microerror.Maskf(invalidExecutionError, "%#q must have at least one item in order to collect metrics for the cordon expiration", datetime)
+	}
+
+	t, err := time.Parse(layout, split[0])
 	if err != nil {
-		return time.Time{}, microerror.Mask(err)
+		return time.Time{}, microerror.Maskf(invalidExecutionError, "unavailable to %#q parsing: %#v", split[0], err.Error())
```

Comment:
What about this? 

```suggestion
		return time.Time{}, microerror.Maskf(invalidExecutionError, "failed to parse %#q", split[0])
```

Reply:
Then it would not contain the reason of why it failed in logging message, which one is better?